scripts/provisioning-tests

```diff
@@ -1,28 +1,80 @@
 #!/bin/bash
-set -e
+set -e -x
+
+cleanup()
+{
+    EXIT=$?
+    set +ex
+    echo Stopping rancher server
+    kill $RANCHER_RUN_PID
+    wait $RANCHER_RUN_PID
+    if [ $PID != -1 ]; then
+      kill $PID
+      wait $PID
+    fi
+    return $EXIT
+}
 
 cd $(dirname $0)/..
 
+SOME_K8S_VERSION=v1.21.5+rke2r2
+TB_ORG=rancher
+
+if [ -z "${DIST}" ]; then
+  DIST=k3s
+  AIRGAP=-airgap
+  SOME_K8S_VERSION=v1.21.5+k3s2
+  TB_ORG=k3s-io
+else
+  LINUX=.linux
+fi
+
+export DIST=${DIST}
+export SOME_K8S_VERSION=${SOME_K8S_VERSION}
+export TB_ORG=${TB_ORG}
+
 eval "$(grep '^ENV CATTLE_SYSTEM_AGENT' package/Dockerfile | awk '{print "export " $2 "=" $3}')"
+eval "$(grep '^ENV CATTLE_KDM_BRANCH' package/Dockerfile | awk '{print "export " $2 "=" $3}')"
+#export SOME_K8S_VERSION=$(curl https://raw.githubusercontent.com/rancher/kontainer-driver-metadata/$CATTLE_KDM_BRANCH/data/data.json | jq -r ".$DIST.channels[0].latest")
 
-echo Starting rancher server
-mkdir -p /var/lib/rancher/k3s/agent/images
-grep PodTestImage ./tests/integration/pkg/defaults/defaults.go | cut -f2 -d'"' > /var/lib/rancher/k3s/agent/images/pull.txt
-grep MachineProvisionImage ./pkg/settings/setting.go | cut -f4 -d'"' >> /var/lib/rancher/k3s/agent/images/pull.txt
+echo Starting rancher server for provisioning-tests
 touch /tmp/rancher.log
+
+mkdir -p /var/lib/rancher/$DIST/agent/images
+grep PodTestImage ./tests/integration/pkg/defaults/defaults.go | cut -f2 -d'"' > /var/lib/rancher/$DIST/agent/images/pull.txt
+grep MachineProvisionImage ./pkg/settings/setting.go | cut -f4 -d'"' >> /var/lib/rancher/$DIST/agent/images/pull.txt
 mkdir -p /usr/share/rancher/ui/assets
 curl -sLf https://github.com/rancher/system-agent/releases/download/${CATTLE_SYSTEM_AGENT_VERSION}/rancher-system-agent-amd64 -o /usr/share/rancher/ui/assets/rancher-system-agent-amd64
-./scripts/run >/tmp/rancher.log 2>&1 &
-PID=$!
 
-keepalive()
+mkdir -p /image-assets
+curl -sLf https://github.com/$TB_ORG/$DIST/releases/download/$SOME_K8S_VERSION/$DIST$AIRGAP-images$LINUX-amd64.tar.gz -o /image-assets/$DIST$AIRGAP-images$LINUX-amd64.tar.gz
+echo Decompressing airgap tarball
+gzip -d /image-assets/$DIST$AIRGAP-images$LINUX-amd64.tar.gz
+
+run_rancher()
```

Comment:
This function is causing a lot of logs in drone. Can we quiet this down?

Reply:
Yes, I removed the `set -x`